l5kit/l5kit/sampling/agent_sampling.py

```diff
@@ -214,7 +214,7 @@ def generate_agent_sample(
         future_tl_faces,
     ) = get_agent_context(state_index, frames, agents, tl_faces, history_num_frames, future_num_frames, )
 
-    if perturbation is not None:
+    if perturbation is not None and len(future_frames) == future_num_frames:
```

Comment:
this is one of the BUG that I found, before we will perturb as long as we have a single future frames, which is very bad. Because you can not recover from perturbation if you do not have enough future frames